alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -262,6 +267,16 @@ func resourceAlicloudEmrClusterCreate(d *schema.ResourceData, meta interface{})
 
 	if chargeType, ok := d.GetOk("charge_type"); ok {
 		request.ChargeType = chargeType.(string)
+		innerPeriod, exist := d.GetOk("period")
+		if chargeType == "PrePaid" {
+			if !exist || innerPeriod.(int) <= 0 {
+				return WrapError(Error("PrePaid chargeType, " +
+					"param period must be needed and greater than 0."))
+			}
+		}
+		if innerPeriod.(int) > 0 {
+			request.Period = requests.NewInteger(innerPeriod.(int))
```

Comment:
I think using a default '1' can replace the checking whether the period value is less than 0.